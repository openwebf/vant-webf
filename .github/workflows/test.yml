name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - dev

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
