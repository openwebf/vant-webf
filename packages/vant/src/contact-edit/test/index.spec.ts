import { VueWrapper } from '@vue/test-utils';
import { ContactEdit, ContactEditInfo } from '..';
import { mount, later } from '../../../test';

const contactInfo = {
  name: 'foo',
  tel: '13000000000',
};

async function submitForm(wrapper: VueWrapper<any>) {
  const form = wrapper.find('form');
  await form.trigger('submit');
  return later();
}

test('should validate contact name before submitting form', async () => {
  const wrapper = mount(ContactEdit, {
    props: {
      contactInfo: {
        name: '',
        tel: '',
      },
    },
  });

  await submitForm(wrapper);
  expect(wrapper.find('.van-field__error-message').html()).toMatchSnapshot();
});

test('should validate contact tel before submitting form', async () => {
  const wrapper = mount(ContactEdit, {
    props: {
      contactInfo: {
        name: 'name',
        tel: '',
      },
    },
  });

  await submitForm(wrapper);
  expect(wrapper.find('.van-field__error-message').html()).toMatchSnapshot();
});

test('should emit save event after submitting form', async () => {
  const wrapper = mount(ContactEdit, {
    props: {
      contactInfo,
    },
  });

  await submitForm(wrapper);
  expect(wrapper.emitted<[ContactEditInfo]>('save')![0][0]).toEqual(
    contactInfo
  );
});

test('should watch contact info', async () => {
  const wrapper = mount(ContactEdit);
  await wrapper.setProps({ contactInfo });
  await submitForm(wrapper);
  expect(wrapper.emitted<[ContactEditInfo]>('save')![0][0]).toEqual(
    contactInfo
  );
});

test('should allow deleting contact', async () => {
  const wrapper = mount(ContactEdit, {
    props: {
      isEdit: true,
    },
  });

  const deleteButton = wrapper.findAll('.van-button')[1];
  deleteButton.trigger('click');

  await later();
  document.querySelector<HTMLElement>('.van-dialog__confirm')?.click();

  await later();
  expect(wrapper.emitted('delete')).toBeTruthy();
});
