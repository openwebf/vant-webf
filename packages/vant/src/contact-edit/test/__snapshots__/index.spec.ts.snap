// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should validate contact name before submitting form 1`] = `
<div class="van-field__error-message">
  Please fill in the name
</div>
`;

exports[`should validate contact tel before submitting form 1`] = `
<div class="van-field__error-message">
  Malformed phone number
</div>
`;
